src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java

```diff
@@ -781,6 +797,48 @@ private static void addPlaces(Leg leg, List<State> states, List<Edge> edges, boo
         }
     }
 
+    private static List<TransitTripStage> generateTransitTripStages(List<State> states) {
+        List<TransitTripStage> transitTripStages = new ArrayList<>();
+
+        int currentTripTime = 1;
+        int firstTransitFareStartsAt = -1;
+        Stop currentStop;
+        double distance;
+
+        for (State currentState : states) {
+            Vertex vertex = currentState.getVertex();
+
+            if (vertex instanceof PatternArriveVertex || vertex instanceof PatternDepartVertex) {
+                currentStop = ((TransitVertex) vertex).getStop();
```

Comment:
Please make here new variable `OnboardVertex currentVertex = (OnboardVertex) vertex;` and use it later without casting.